test/hpke_test.c

```diff
@@ -1784,6 +1779,83 @@ static int test_hpke_ikms(void)
     return res;
 }
 
+/*
+ * Test that use of a compressed format auth public key works
+ * We'll do a typical round-trip for auth mode but provide the
+ * auth public key in compressed form. That should work.
+ */
+static int test_hpke_compressed()
+{
+    int erv = 0;
+    EVP_PKEY *privp = NULL;
+    unsigned char pub[OSSL_HPKE_TSTSIZE];
+    size_t publen = sizeof(pub);
+    EVP_PKEY *authpriv = NULL;
+    unsigned char authpub[OSSL_HPKE_TSTSIZE];
+    size_t authpublen = sizeof(authpub);
+    int hpke_mode = OSSL_HPKE_MODE_AUTH;
+    OSSL_HPKE_SUITE hpke_suite = OSSL_HPKE_SUITE_DEFAULT;
+    OSSL_HPKE_CTX *ctx = NULL;
+    OSSL_HPKE_CTX *rctx = NULL;
+    unsigned char plain[] = "quick brown fox";
+    size_t plainlen = sizeof(plain);
+    unsigned char enc[OSSL_HPKE_TSTSIZE];
+    size_t enclen = sizeof(enc);
+    unsigned char cipher[OSSL_HPKE_TSTSIZE];
+    size_t cipherlen = sizeof(cipher);
+    unsigned char clear[OSSL_HPKE_TSTSIZE];
+    size_t clearlen = sizeof(clear);
+
+    hpke_suite.kem_id = OSSL_HPKE_KEM_ID_P256;
+
+    /* generate auth key pair */
+    if (!TEST_true(OSSL_HPKE_keygen(hpke_suite, authpub, &authpublen, &authpriv,
+                                    NULL, 0, testctx, NULL)))
+        goto end;
+    /* now get the compressed form public key */
+    if (!TEST_true(EVP_PKEY_get_octet_string_param(authpriv,
+                                                   OSSL_PKEY_PARAM_PUB_KEY,
```

Comment:
We cannot rely on OSSL_PKEY_PARAM_PUB_KEY always being returned in compressed format. It happens to be that way right now, but the documentation doesn't guarantee it and it is considered a bug that it doesn't respect the default "point conversion format". There are moves afoot to change it to respect the point conversion format which defaults to uncompressed (see #19681). Probably you need to call `EVP_PKEY_set_utf8_string_param()` first to set `OSSL_PKEY_PARAM_EC_POINT_CONVERSION_FORMAT` to "compressed". Otherwise, as soon as #19681 is merged, it won't be testing what we think its testing any more.

Reply:
thanks - added that (though the loooong names probably cause new tab-alignment nits;-)

Reply:
done in https://github.com/openssl/openssl/pull/17172/commits/b38d5a27c0acae15aa9e87595dea2e09451cebc3